### PR TITLE
Workflow.concurrency and Job.concurrency

### LINKED ISSUE #119: [Core feature request] concurrency
**What feature do you need?**

- https://docs.github.com/en/actions/using-jobs/using-concurrency
- https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idconcurrency

**Do you have an example usage?**

```yaml
concurrency: staging_environment
```

```yaml
concurrency: 
  group: ${{ github.ref }}
  cancel-in-progress: true
```

Keep in mind that it's available both
- at a workflow level
- at `jobs.<job_id>.concurrency` level

### PR DESCRIPTION
**Pull request by jmfayard**
Resolves #119

### CODE REVIEW

File: library/src/main/kotlin/it/krzeminski/githubactions/dsl/Concurrency.kt [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package it.krzeminski.githubactions.dsl
```
Comment (krzema12) [inline library/src/main/kotlin/it/krzeminski/githubactions/dsl/Concurrency.kt:1]: It should rather be in the domain package:
```suggestion
package it.krzeminski.githubactions.domain
```

File: library/src/main/kotlin/it/krzeminski/githubactions/dsl/Concurrency.kt [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+
+/**
+ * Using concurrency allow to run a workflow/job at a time
```
Comment (krzema12) [inline library/src/main/kotlin/it/krzeminski/githubactions/dsl/Concurrency.kt:7]: Minor grammar issues:
```suggestion
 * Using concurrency allows running single workflow/job at a time
```
does it still preserve the meaning it should, after my fixes? :)

File: library/src/main/kotlin/it/krzeminski/githubactions/yaml/ToYaml.kt [hunk 3]
```diff
@@ -70,6 +70,12 @@ private fun Workflow.generateYaml(addConsistencyCheck: Boolean, useGitDiff: Bool
         appendLine()
         appendLine("name: $name")
         appendLine()
+        if (concurrency != null) {
```
Comment (krzema12) [inline library/src/main/kotlin/it/krzeminski/githubactions/yaml/ToYaml.kt:73]: Could you move it below the triggers ("on:")? It seems to be less important.

**Review by krzema12 — CHANGES_REQUESTED** [review]

File: library/src/main/kotlin/it/krzeminski/githubactions/dsl/Concurrency.kt (hunk 1 above)
Comment (jmfayard) [reply library/src/main/kotlin/it/krzeminski/githubactions/dsl/Concurrency.kt]: done

File: library/src/main/kotlin/it/krzeminski/githubactions/yaml/ToYaml.kt (hunk 3 above)
Comment (jmfayard) [reply library/src/main/kotlin/it/krzeminski/githubactions/yaml/ToYaml.kt:77]: done

File: library/src/main/kotlin/it/krzeminski/githubactions/dsl/Concurrency.kt (hunk 2 above)
Comment (jmfayard) [reply library/src/main/kotlin/it/krzeminski/githubactions/dsl/Concurrency.kt]: yes it does :)

**Comment by jmfayard** [comment]
I'm done

**Review by krzema12 — APPROVED** [review]

**Comment by krzema12** [comment]
Thank you! :guitar:

**Pull request merged**